addons/mail/static/src/models/composer/composer.js

```diff
@@ -287,9 +298,48 @@ function factory(dependencies) {
             body = this._generateMentionsLinks(body);
             body = parseAndTransform(body, addLink);
             body = this._generateEmojisOnHtml(body);
+            return body;
+        }
+
+        /**
+         * Recursive function that process the queue.
+         */
+        async processMessageQueue() {
+            if (!this.isProcessingQueue && !this.isQueueError) {
+                try {
+                    this.update({
+                        isProcessingQueue: true,
+                    });
+                    const message = this.thread.queue[0];
+                    await this._postNextMessageInQueued(message);
+                    this.thread.update({ queue: [['unlink', message]] });
+                } catch(error) {
+                    this.update({
+                        isQueueError: true,
+                    });
+                } finally {
+                    this.update({
+                        isProcessingQueue: false,
+                    });
+                    if (this.thread.queue.length > 0) {
+                        this.processMessageQueue();
+                    } else {
+                        this.thread.refreshFollowers();
+                        this.thread.fetchAndUpdateSuggestedRecipients();
```

Comment:
probably don't need to refresh if there is a crash? so just have this inside try and not inside finally??